### ISSUE TITLE
Invalid arguments to printf %(datefmt)T not detected

### ISSUE THREAD

**Issue by HaleTom** [koalaman/shellcheck#886]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
printf -v time "%(%b %d %H:%M)T" foo
echo "$time"
```

#### Here's what shellcheck currently says:
`No issues detected!`

#### Here's what I wanted or expected to see:
Invalid parameter to datefmt: Expecting an integer greater than or equal to -2.

--------

From the `bash` manual:

> %(datefmt)T
>  causes printf to output the date-time string resulting from using datefmt as a format string for strftime(3).  The corresponding argument is an integer representing                     the  number  of seconds since the epoch.  Two special argument values may be used: -1 represents the current time, and -2 represents the time the shell was invoked.                      If no argument is specified, conversion behaves as if -1 had been given.  This is an exception to the usual printf behavior.

**Comment by aw-was-here** [comment]
shellcheck's handling of bash 4.2+'s date handling is a bit unpredictable:

printf "%(%s)T" throws a SC2183 because it thinks the %s is a string parameter.

**Comment by koalaman** [comment]
@aw-was-here ShellCheck understands this one, but is giving a SC2183 because you're specifying a date format but no argument to format accordingly. Here's the wiki:

>When using the Ksh/Bash %T timestamp extension, such as printf 'The time is %(%H:%M)T\n', an argument of -1 and no argument are both taken to mean the current time. In these cases, consider specifying -1 explicitly.
